### PR TITLE
Fixes #9315 - Add examples to range operator

### LINKED ISSUE #9315: Add a warning box to the range operator, explaining the necessities for applying it to strings
### Prerequisites

- [X] **Existing Issue:** Search the existing issues for this repository. If there is an issue that fits your needs do not file a new one. Subscribe, react, or comment on that issue instead.
- [X] **Descriptive Title:** Write the title for this issue as a short synopsis. If possible, provide context. For example, "Typo in `Get-Foo` cmdlet" instead of "Typo."
- [X] **Verify Version:** If there is a mismatch between documentation and the behavior on your system, ensure that the version you are using is the same as the documentation. Check this box if they match or the issue you are reporting is not version specific.

### Links

https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_operators#range-operator-

### Summary

Currently, the documentation on the `range` operator does not explain what happens behind the curtains.

So, It's hard for the user to understand what the result of the range operator actually is. This is particularly true when the `range` operator is being applied to characters/string.

### Details

See [this PowerShell issue](/PowerShell/PowerShell/issues/18309) for reference.

### Suggested Fix

I propose to add a warning box to the `range` operator,<br/>informing the user that …

1. It creates an **_array_** of integers or characters.
2. When assigned to a string, a "character array to string" conversion takes place, putting spaces between each character. There is an $OFS preference variable controlling this behaviour.
3. Add a link to that $OFS preference variable.
4. Add an example of how to use the `range` operator in an "character array to string" conversion.

### PR DESCRIPTION
**Pull request by sdwheeler**
# PR Summary

Fixes [AB#20534](https://dev.azure.com/msft-skilling/cebd7ef5-4282-448b-9701-88c8637581b7/_workitems/edit/20534) - Fixes #9315 - Add examples to range operator

## PR Checklist

- [x] **Descriptive Title:** This PR's title is a synopsis of the changes it proposes.
- [x] **Summary:** This PR's summary describes the scope and intent of the change.
- [x] **Contributor's Guide:** I have read the [contributors guide][contrib].
- [x] **Style:** This PR adheres to the [style guide][style].

[contrib]: https://learn.microsoft.com/powershell/scripting/community/contributing/overview
[style]: https://learn.microsoft.com/powershell/scripting/community/contributing/powershell-style-guide

### CODE REVIEW

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Operators.md [hunk 1]
```diff
@@ -444,6 +449,35 @@ You can also create ranges in reverse order.
...
+6
+7
+8
+9
+10
+```
+
+> [!IMPORTANT]
+> The resulting range isn't limited to the values of the enumeration. Instead
+> is represents the range of values between the two values provided. You can't
```
Comment (michaeltlombardi) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Operators.md:478]:
```suggestion
> it represents the range of values between the two values provided. You can't
```

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Operators.md [hunk 2]
```diff
@@ -429,11 +429,16 @@ Get-Service | Where-Object {$_.StartType -eq 'Automatic'}
...
 ### Range operator `..`
 
-Represents the sequential integers in an integer array, given an upper, and
-lower boundary.
+The range operator can be used to represents an array of sequential integers.
+The values joined by the range operator define the start and end values of the
+range.
+
+> [!NOTE]
+> Support for character ranges was added in PowerShell 6.
```
Comment (michaeltlombardi) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Operators.md:437]: Do we normally note support for features/behaviors in future versions?

File: reference/7.0/Microsoft.PowerShell.Core/About/about_Operators.md [hunk 3]
```diff
@@ -554,6 +556,71 @@ B
...
+6
+7
+8
+9
+10
+```
+
+> [!IMPORTANT]
+> The resulting range isn't limited to the values of the enumeration. Instead
+> is represents the range of values between the two values provided. You can't
```
Comment (michaeltlombardi) [inline reference/7.0/Microsoft.PowerShell.Core/About/about_Operators.md:621]:
```suggestion
> it represents the range of values between the two values provided. You can't
```

File: reference/7.0/Microsoft.PowerShell.Core/About/about_Operators.md [hunk 4]
```diff
@@ -554,6 +556,71 @@ B
 A
 ```
 
+If you assign a character range to a string, it's treated the same assigning a
+character array to a string.
```
Comment (michaeltlombardi) [inline reference/7.0/Microsoft.PowerShell.Core/About/about_Operators.md:560]: Is it worth noting that specifying non-character strings throws an exception? I verified in a try/catch that it does.

File: reference/7.0/Microsoft.PowerShell.Core/About/about_Operators.md [hunk 5]
```diff
@@ -554,6 +556,71 @@ B
...
+$a
+a b c d e
+```
+
+The characters in the array are joined into a string. The characters are
+separated by the value of the `$OFS` preference variable. For more information,
+see [about_Preference_Variables](about_Preference_Variables.md#ofs).
+
+The order of the characters in the array is determined by the ASCII value of
+the character. For example, the ASCII values of `c` and `X` are 99 and 88,
```
Comment (michaeltlombardi) [inline reference/7.0/Microsoft.PowerShell.Core/About/about_Operators.md:576]: Is it worth linking to [this page for the character codes][01] or mentioning the usability of the [GetBytes][02] method to retrieve the value of a given character, like:

```powershell
[System.Text.ASCIIEncoding]::ASCII.GetBytes(5)
[System.Text.ASCIIEncoding]::ASCII.GetBytes(f)
```

```output
53
102
```

[01]: https://learn.microsoft.com/office/vba/language/reference/user-interface-help/character-set-0127
[02]: https://learn.microsoft.com/dotnet/api/system.text.asciiencoding.getbytes

File: reference/7.3/Microsoft.PowerShell.Core/About/about_Operators.md [hunk 6]
```diff
@@ -554,6 +556,71 @@ B
...
+6
+7
+8
+9
+10
+```
+
+> [!IMPORTANT]
+> The resulting range isn't limited to the values of the enumeration. Instead
+> is represents the range of values between the two values provided. You can't
```
Comment (michaeltlombardi) [inline reference/7.3/Microsoft.PowerShell.Core/About/about_Operators.md:621]:
```suggestion
> it represents the range of values between the two values provided. You can't
```

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Operators.md (hunk 1 above)
Comment (michaeltlombardi) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Operators.md:478]:
```suggestion
> it represents the range of values between the two values provided. You can't
```

File: reference/7.2/Microsoft.PowerShell.Core/About/about_Operators.md [hunk 7]
```diff
@@ -554,6 +556,71 @@ B
...
+6
+7
+8
+9
+10
+```
+
+> [!IMPORTANT]
+> The resulting range isn't limited to the values of the enumeration. Instead
+> is represents the range of values between the two values provided. You can't
```
Comment (michaeltlombardi) [inline reference/7.2/Microsoft.PowerShell.Core/About/about_Operators.md:621]:
```suggestion
> it represents the range of values between the two values provided. You can't
```

**Review by michaeltlombardi — CHANGES_REQUESTED** [review]
Minor typo and some questions.

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Operators.md (hunk 2 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Operators.md:437]: I meant to remove that.

File: reference/7.0/Microsoft.PowerShell.Core/About/about_Operators.md (hunk 4 above)
Comment (sdwheeler) [reply reference/7.0/Microsoft.PowerShell.Core/About/about_Operators.md:560]: I don't think that is necessary.

File: reference/7.0/Microsoft.PowerShell.Core/About/about_Operators.md (hunk 5 above)
Comment (sdwheeler) [reply reference/7.0/Microsoft.PowerShell.Core/About/about_Operators.md:576]: Don't want to complicate things. Plus the simpler form I was using to test was `[byte]([char]'X')`.

**Review by michaeltlombardi — APPROVED** [review]

**Comment by SetTrend** [comment]
Marvellous!!

I just found a minor typo in the first file (the type occurs in several locations in the file):

>  The range operator can be used to represents …

**Pull request merged**